ToolBox/Visualizer/QuantConnect.Visualizer.py

```diff
@@ -0,0 +1,182 @@
+        with open('config.json', 'r') as json_data:
+            self.config = json.load(json_data)
+
+        # Loads the Toolbox to access Visualizer
+        self.setup_and_load_toolbox()
+
+        # Get plot image name, if not defined, generate one.
+        self.plot_filename = arguments['--output']
+        if self.plot_filename is None:
+            self.plot_filename = self.generate_plot_filename()
+
+        self.size_px = [int(p) for p in arguments['--size'].split(',')]
+
+    def setup_and_load_toolbox(self):
+        """
+        Checks if the path given in the config.json file contains the needed assemblies, if so it returns the
+        absolute path to that folder.
+        Also, saves a copy of the config.json with the data needed by the Composer to run.
+        :return: void.
+        :raise: NotImplementedError: if the needed assemblies dll are not available.
+        """
+        # Check Lean assemblies are present in the composer-dll-directory key provided.
+        assemblies_folder_info = (Path(self.config['composer-dll-directory'])
+                                  if 'composer-dll-directory' in self.config else Path('../'))
+        if not assemblies_folder_info.joinpath('QuantConnect.ToolBox.exe').exists():
+            raise KeyError(
+                "Please set up the 'composer-dll-directory' key in config.json with the path to Lean assemblies." +
+                f"Absolute path to 'composer-dll-directory' provided: {assemblies_folder_info.resolve().absolute()}\n" +
+                'If the issue continues even then the folder is correctly declared, please use the --reset option.')
+        # Check Data folder is correctly set up
+        data_folder_info = (
+            Path(self.config['data-folder']) if 'data-folder' in self.config else Path('../../../Data/'))
+        if not data_folder_info.joinpath('market-hours', 'market-hours-database.json').exists():
+            raise KeyError("Please set up a valid 'data-folder' key in config.json.\n" +
+                           f"Absolute path to 'data-folder' provided: {data_folder_info.resolve().absolute()}\n" +
+                           'If the issue continues even then the folder is correctly declared, please use the --reset option.')
+        assembly_folder_path = str(assemblies_folder_info.resolve().absolute())
+        # Check if config.json exist in the composer-dll-directory, if not creates it.
+        config_file = assemblies_folder_info.joinpath('config.json')
+        if not config_file.exists() or self.reset:
+            cfg_content = {'composer-dll-directory': assembly_folder_path,
+                           'data-folder': str(data_folder_info.resolve().absolute())}
+            with open(str(config_file.resolve().absolute()), 'w') as cfg:
+                json.dump(cfg_content, cfg)
+
+        # Load the visualizer
+        os.chdir(assembly_folder_path)
+        sys.path.append(assembly_folder_path)
+        AddReference("QuantConnect.ToolBox")
```

Comment:
You don't need to `chdir` and add the assembly folder to path, in `AddReference` you can put `QuantConnect.ToolBox.exe` full file name.